### PR TITLE
[draft] modernize babel configs + modern core

### PR DESCRIPTION
**Pull request by developit**

### CODE REVIEW

File: packages/next/taskfile.js [hunk 1]
```diff
@@ -4,33 +4,13 @@ const relative = require('path').relative
...
-        modules: 'commonjs',
-        targets: {
-          esmodules: true,
-        },
-        loose: true,
-        exclude: ['transform-typeof-symbol'],
-      },
-    ],
-    '@babel/preset-react',
+    ['@babel/preset-react', { loose: true, useBuiltIns: true }],
```
Comment (Timer) [inline packages/next/taskfile.js]: How about `useSpread` instead of `useBuiltIns`?

Comment (developit) [reply packages/next/taskfile.js]: Good call. Babel merged this into the JSX transform, and [useSpread](https://babeljs.io/docs/en/babel-plugin-transform-react-jsx#usespread) is the name they ended up going with.

Comment (timneutkens) [reply packages/next/taskfile.js]: At this point in time `useBuiltIns` is better as it results in `Object.assign ` whereas spread ends up being a helper, currently investigating if we can change that to also compile to Object.assign in loose mode.

**Comment by timneutkens** [comment]
Closing this as the PR is outdated

**Pull request closed without merging**